### PR TITLE
Use a modal for license explanation

### LINKED ISSUE #3582: Modal popover for report and license definition
## Problem
As reported in #569, reporting content is currently made from a popover that looks broken on small screens. Based on the recent Design Library updates, we can improve this, and also the license definition popover, by reusing the component merged in #2860

### Issues related

- Report of a broken popover in #569
- The basis for a future suggested in #645
- Modal specs in #2860

### Search results page

Since the license description element is toggled from the drawer in small screens, I have two ideas that I would like your thoughts on.

#### v1. Replace drawer with the license

In this first version, once you click on the license’s question icon, the drawer disappears and is replaced by the license popover. You can see how the results grid is visible behind the modal layer.

Drawer  v1

#### v2. Push the drawer behind modal opaque layer 

The second idea is to push the drawer behind the modal layer to avoid losing the context of where users displayed the info from. See how the drawer remains behind the dark layer.

I am more drawn to this version as contexts in modals play a key role.

Drawer  v2

#### From `LG` and beyond

On bigger screens, the license description popover is displayed as in the current flow.

Results with modal popover

### Single result page

In the single result page, the popover shows up as in the current flow.

Single result page with modal popover

## Popovers

As described in #2860, the following specs are still valid:

- Popover is centered, full width, and has a max value of 400px.
- On `xs` , horizontal margins `24px`

However, I noticed that we did not consider a scrollable popover for those cases where its height overflows the device screen. For that case, I tried the following:

Scrollable popover behavior

The spacings mirror the ones applied to the current component, but instead of wrapping the content with a single element, the popover's header is separated to keep it sticky when scrolling down. Below a screenshot of dev mode.

Popover with notes in header

## Mockups

Figma file: [Modal popover for report and license definition](https://www.figma.com/file/chRWINXqyRzYRPZOJOJHuD/Modal-popover-for-report-and-license-definition?type=design&node-id=22-13656&mode=dev)

### LINKED ISSUE #4802: Big white space in the modal of content report
## Description

There is an unnecessary white space in the top side of the content area in the recent implementation of the report content as modal (#4710 to solve #3582).

Solution: the spacing should be zero.

## Screenshots

| Page | Page with inspector |
|:---|:---|
| Screenshot of content report modal | Screenshot of content report modal with arrow pointing the padding top of content area |

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #3582 by @fcoveram
Fixes #4802 by @fcoveram 

## Description

This PR converts the license explanation popover to a modal.

### Desktop

Screenshot 2024-08-23 at 11 43 00 AM

### Mobile
Screenshot 2024-08-23 at 11 43 41 AM

### Functionality changes
On mobile screens, the license explanation modal appears on top of the content settings modal. So, when the user clicked outside of the license explanation modal, the lower content settings modal handled this event and closed, keeping the top-most license explanation one open.
To correctly handle this, I added a `use-dialog-stack` composable and `id`s to all modals and popovers to keep track and route the event to the currently open, top-most dialog.

### Vertical overflow
I added the `max-height` to modals, set to the full height of the screen, minus the height of the modal header, minus an arbitrary value of 6rem. @fcoveram, what is the best value for vertical margin of the modal?

Screenshot 2024-08-23 at 11 47 30 AM

## Testing Instructions

Run the app and go to `http://localhost:8443/search?q=cat`.
On desktop, open the filters sidebar and click on one of the `?` buttons next to licenses. The modal should appear. It's styles should match the mockups, and it should function correctly.
On a mobile, open the content settings modal, select the _Filters_ tab and click on one of the `?` buttons next to licenses. The modal should appear. Clicking on the close button, pressing <kbd>Escape</kbd> or clicking outside of the modal should close it, and only it (not the content settings modal below it).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VContentReport/VContentReportModal.vue [hunk 1]
```diff
@@ -51,7 +52,7 @@ const close = () => {
     <template #default>
       <VContentReportForm
         ref="contentReportFormRef"
-        class="p-7 pt-0 sm:p-9"
+        class="p-7 pt-0 sm:p-9 sm:pt-0"
```
Comment (obulat) [inline frontend/src/components/VContentReport/VContentReportModal.vue:55]: This is the fix for #4802

File: frontend/src/components/VPopover/VPopover.vue [hunk 2]
```diff
@@ -15,6 +15,7 @@ import type { Placement, Strategy } from "@floating-ui/dom"
 
 const props = withDefaults(
   defineProps<{
+    id: string
```
Comment (zackkrida) [inline frontend/src/components/VPopover/VPopover.vue:18]:
```suggestion
    /**
     * The id used to keep track of the modal in the open modal stack, to enable
     * nested modals.
     */
    id: string
```

Should this have a similar comment?

File: frontend/src/composables/use-dialog-stack.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  return {
+    stack,
+    push,
+    pop,
+    clear,
+    indexOf,
+    activeDialog,
+  }
+}
```
Comment (zackkrida) [inline frontend/src/composables/use-dialog-stack.ts:29]: Can we document what activeDialog means in terms of the actual UI here? For example saying it's the "parent dialog which other dialogs can appear on top of but has no dialogs below it" or something. That's a _terrible_ example but hopefully conveys what I mean.

**Review by zackkrida — APPROVED** [review]
I love the ID stack approach to fixing the nested dialog issue! Everything I've tested works great. Some documentation nits, otherwise LGTM.

**Pull request merged**